src/Plugin/facets/facet_source/ListFacetSource.php

```diff
@@ -0,0 +1,127 @@
+<?php
+
+declare(strict_types = 1);
+
+namespace Drupal\oe_list_pages\Plugin\facets\facet_source;
+
+use Drupal\Core\Path\CurrentPathStack;
+use Drupal\facets\FacetSource\SearchApiFacetSourceInterface;
+use Drupal\facets\Plugin\facets\facet_source\SearchApiBaseFacetSource;
+use Drupal\search_api\Entity\Index;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+use Symfony\Component\HttpFoundation\RequestStack;
+
+/**
+ * Represents a facet source to be used for lists.
+ *
+ * @FacetsFacetSource(
+ *   id = "list_facet_source",
+ *   display_id = "oe_list_pages",
+ *   label = "OE List Pages",
+ *   deriver = "Drupal\oe_list_pages\Plugin\facets\facet_source\ListFacetSourceDeriver"
+ * )
+ */
+class ListFacetSource extends SearchApiBaseFacetSource implements SearchApiFacetSourceInterface {
+
+  /**
+   * The current path stack.
+   *
+   * @var \Drupal\Core\Path\CurrentPathStack
+   */
+  protected $currentPathStack;
+
+  /**
+   * The request stack.
+   *
+   * @var \Symfony\Component\HttpFoundation\RequestStack
+   */
+  protected $requestStack;
+
+  /**
+   * {@inheritdoc}
+   */
+  public function __construct(array $configuration, $plugin_id, array $plugin_definition, $query_type_plugin_manager, $search_results_cache, RequestStack $request_stack, CurrentPathStack $current_path_stack) {
+    parent::__construct($configuration, $plugin_id, $plugin_definition, $query_type_plugin_manager, $search_results_cache);
+
+    $this->currentPathStack = $current_path_stack;
+    $this->requestStack = $request_stack;
+    $index = $plugin_definition['index'];
+    $this->index = Index::load($index);
```

Comment:
Why do you need to do this in the constructor? Not all instantiations will need the index loaded. Just load it inside the getIndex() method when it's first requested. Also, inject the entity type manager.